src/nupic/algorithms/Connections.cpp

```diff
@@ -199,17 +225,33 @@ void Connections::destroySynapse(Synapse synapse) {
   segmentData.synapses.erase(synapseOnSegment);
 
   destroyedSynapses_.push_back(synapse);
+
+  if( synapseData.permanence > connectedThreshold_ - EPSILON )
```

Comment:
can we do this in constructor once, and then used the (modified) internal variable connectedThreshold_ ? 
```
initialize(...) {
  connectedThresh_ = connectedThreshold - EPSILON; 
  NTA_CHECK(...);
}

getConnectedThreshold() { return connectedThreshold_ + EPSILON;} 
```

benefits: 
- "faster" code (no need for computing the minus in the if condition
- NTA_CHECK actually works better, as tests the really used value